### PR TITLE
feat: delta rule work with zero length sequence

### PR DESCRIPTION
**Pull request by guangyunh-nv**
## 📌 Description

Avoid current sm100 gdn to deadlock on inputs with sequence length as 0 

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

## Reviewer Notes

This is a very edge case we have ever met in production environment from customers for softmax attention.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed kernel handling of empty or zero-length sequences to prevent undefined behavior and ensure correct state management.

* **Tests**
  * Added test coverage for zero-length sequence scenarios in the Gated Delta Net kernel.

### CODE REVIEW

File: tests/gdn/test_prefill_delta_rule.py [hunk 1]
```diff
@@ -259,6 +259,77 @@ def test_prefill_kernel_nonfull(
...
+        scale,
+        None,
+        False,
+        cu_seq_lens_with_empty,
+        True,
+        output=our_o,
+    )
+    torch.cuda.synchronize()
+
+    torch.testing.assert_close(our_o, ref_o, atol=2e-2, rtol=2e-2)
```
[the remark this answers is not in the document]
Comment (guangyunh-nv) [reply tests/gdn/test_prefill_delta_rule.py:330]: Just verify zero length seqs are correctly skipped and do not cause deadlock is enough.

**Review by jiahanc — APPROVED** [review]
LGTM, thanks for contribution!

File: flashinfer/gdn_kernels/blackwell/gated_delta_net_chunked.py [hunk 2]
```diff
@@ -1289,9 +1265,34 @@ def _cg(num_threads):
                         state_inp_ready_producer,
                         shared_inp_ready_producer,
                         o_store_producer,
-                    ),
-                    (True, 0, head_idx),
-                )
+                        checkpoint_idx,
```
[the remark this answers is not in the document]
Comment (guangyunh-nv) [reply flashinfer/gdn_kernels/blackwell/gated_delta_net_chunked.py:1268]: This is only indent change.

File: flashinfer/gdn_kernels/blackwell/gated_delta_net_chunked.py [hunk 3]
```diff
@@ -1235,49 +1235,25 @@ def _cg(num_threads):
...
-                    (tiled_mma_kv, tiled_mma_qs, tiled_mma_qkv),
-                    (
-                        sV_pisl,
-                        sCumsumlog,
-                        sCumprod,
-                        sBeta,
-                        sO_pisl,
-                    ),
-                    (mS_checkpoints, checkpoint_offset, checkpoint_every_n_tokens),
+                if num_chunks_b > 0:
```
[the remark this answers is not in the document]
Comment (guangyunh-nv) [reply flashinfer/gdn_kernels/blackwell/gated_delta_net_chunked.py:1244]: This is just for work around cutedsl compilation issue.
